pkg/execute/executor.go

```diff
@@ -220,7 +223,9 @@ func (e *DefaultExecutor) Execute(ctx context.Context) interactive.Message {
 	cmdVerb := strings.ToLower(args[0])
 	var cmdRes string
 	if len(args) > 1 {
-		cmdRes = strings.ToLower(args[1])
+		if !strings.Contains(args[1], "--cluster-name=") {
```

Comment:
@mszostok any tips on how to support parameters like here?
`@botkube help`  -  `args[0]="help", args[1]=""`
`@botkube help --cluster-name=c1`  -  `args[0]="help", args[1]!=""`

Reply:
yep, this is tricky. In general, we should get rid of the`--cluster-name` flag once we will remove the legacy Slack: https://github.com/kubeshop/botkube/issues/865


However, for now we need to handle that. What's more, you need to handle:

- `@botkube help --cluster-name=foo`
- `@botkube help --cluster-name foo` // no `=` sign
- `@botkube kc get po --cluster-name    foo    -n default` // whitespace, additional flag after

Currently, we have 2 approaches in our code base:
1. Old approach still used in `kubectl` executor, where the `--cluster-name` is removed in the for loop: https://github.com/kubeshop/botkube/blob/22da62e1a67a85e0fb2e4af9a6e9106871179d14/pkg/execute/kubectl.go#L210-L243
2. New approach which is more reliable where `cobra` flag parser is used and later flag is removed from a given string: https://github.com/kubeshop/botkube/blob/22da62e1a67a85e0fb2e4af9a6e9106871179d14/pkg/execute/executor_filter.go#L105-L135


IMO you can copy and adapt the 2nd approach for cluster-name and simply remove it from the command string.

Reply:
@mszostok Just a minor comment about

> In general, we should get rid of the--cluster-name flag once we will remove the legacy Slack

I don't think we can do this, if multi-cluster is also supported by Mattermost and Discord. Or, do you mean enforcing users to create a dedicated app per cluster for all comm platforms? If that's not a common use-case, then maybe it makes sense 🤔 to be discussed later, I put it in the https://github.com/kubeshop/botkube/issues/865